RELEASE_CHECKLIST.md

```diff
@@ -0,0 +1,153 @@
+
+This is a list of the things that need to happen during a release.
+
+## Build a Release
+
+### Prepare the Changelog (Full release only)
+
+1. Open the associated milestone. All issues and PRs should be closed. If
+   they are not you should reassign all open issues and PRs to future
+   milestones.
+1. Go through the commit history since the last release. Ensure that all PRs
+   that have landed are marked with the milestone. You can use this to
+   show all the PRs that are merged on or after YYY-MM-DD:
+   `https://github.com/issues?utf8=%E2%9C%93&q=repo%3Aapollographql%2Fapollo-cli+merged%3A%3E%3DYYYY-MM-DD`
+1. Go through the closed PRs in the milestone. Each should have a changelog
+   label indicating if the change is documentation, feature, fix, or maintenance. If
+   there is a missing label, please add one. If it is a breaking change, also add a changelog - BREAKING label.
+1. Choose an emoji for the release. Try to make it semi-related to something that's been included in the release.
+1. Add this release to the `CHANGELOG.md`. Use the structure of previous
+   entries. If you are releasing a beta or a release candidate, no official changelog is needed, but you're not off the hook! You'll need to write testing instructions in lieu of an official changelog.
+
+### Update cargo manifest
+
+1. Update the version in `Cargo.toml`.
+1. Run `cargo update`.
+1. Run `cargo test`.
+1. Run `cargo build`.
+
+### Update npm manifest and assets
+
+1. Copy `README.md` to `npm/README.md`
+1. Bump the version number in `npm/package.json`
+1. `cd npm && npm install` _Note: This step will appear to fail, however its utility is re-building npm-shrinkwrap.json_
+
+### Start a release PR
+
+1. Create a new branch "#.#.#" where "#.#.#" is this release's version (release) or "#.#.#-rc.#" (release candidate)
+1. Push up a commit with the `Cargo.toml`, `Cargo.lock` and `CHANGELOG.md` changes. The commit message can just be "#.#.#" (release) or "#.#.#-rc.#" (release candidate)
+1. Request review from the Apollo GraphQL tooling team.
+
+### Review
+
+Most review comments will be about the changelog. Once the PR is finalized and approved:
+
+1. If you made changes, squash or fixup all changes into a single commit.
+1. Run `git push` and wait for CI to pass.
+
+### Tag and build release
+
+This part of the release process is handled by GitHub Actions, and our binaries are distributed as GitHub Releases. When you push a version tag, it kicks off an action that creates a new GitHub release for that tag, builds release binaries and attaches them to the release.
+
+1. Once ready to merge, tag the commit by running either `git tag -a v#.#.# -m #.#.#` (release), or `git tag -a v#.#.#-rc.# -m #.#.#` (release candidate)
+1. Run `git push --tags`.
+1. Wait for CI to pass.
+
+### Edit the release
+
+After CI builds the release binaries and they appear on the [releases page](https://github.com/apollographql/apollo-cli/releases), click `Edit` and update release notes.
```

Comment:
Is this needed with the changelog being added? Or is this for additional information?

Reply:
So right now the release CI does not automatically take the changelog and put it into the release. It's a manual step for now but once we hammer down how we're formatting our changelogs I'm very happy to put in some time to automate it.